server/src/main/java/io/crate/execution/ddl/tables/AlterTableTask.java

```diff
@@ -35,13 +38,22 @@ public class AlterTableTask<T> extends DDLClusterStateTaskExecutor<T> {
     private final NodeContext nodeContext;
     private final RelationName relationName;
     private final AlterTableOperator<T> alterTableOperator;
+    private final FulltextAnalyzerResolver fulltextAnalyzerResolver;
 
     public AlterTableTask(NodeContext nodeContext,
                           RelationName relationName,
                           AlterTableOperator<T> alterTableOperator) {
+        this(nodeContext, relationName, alterTableOperator, null);
+    }
+
+    public AlterTableTask(NodeContext nodeContext,
+                          RelationName relationName,
+                          AlterTableOperator<T> alterTableOperator,
+                          FulltextAnalyzerResolver fulltextAnalyzerResolver) {
         this.nodeContext = nodeContext;
         this.relationName = relationName;
         this.alterTableOperator = alterTableOperator;
+        this.fulltextAnalyzerResolver = fulltextAnalyzerResolver;
```

Comment:
Can we either bake the `FulltextAnalyzerResolver` dependency into the `AlterTableOperator` or make it required and remove the second constructor?

Otherwise it is a bit error prone as there is no clear contract between which constructor works for which `AlterTableOperator`